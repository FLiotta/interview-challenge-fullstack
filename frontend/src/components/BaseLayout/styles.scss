.base-layout {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  background-color: #FDFCFF;

  &__body {
    padding: 25px;
  }
}