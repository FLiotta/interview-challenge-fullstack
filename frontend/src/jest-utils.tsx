// @Packages
import { ReactNode } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';

// @Project
import store from 'store';
import { JsxElement } from 'typescript';

function render(
  ui: ReactNode | JsxElement | any,
  renderOptions: any = { }
) {
  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>{children}</Provider>
    )
  }
  return rtlRender(ui, {
     wrapper: Wrapper, 
     ...renderOptions
    })
}

export * from '@testing-library/react'

export { render }
