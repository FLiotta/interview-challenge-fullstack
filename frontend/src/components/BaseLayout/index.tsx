// @Project
import Navbar from 'components/Navbar';
import { JsxElement } from 'typescript';

// @Own
import './styles.scss';

interface IProps {
  children: JsxElement | React.ReactNode | React.ReactElement
}

const BaseLayout: React.FC<IProps> = ({ children }) => (
  <div className="base-layout">
    <Navbar />
    <div className="base-layout__body">
      {children}
    </div>
  </div>
)

export default BaseLayout;