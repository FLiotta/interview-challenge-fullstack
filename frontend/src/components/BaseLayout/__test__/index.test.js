// @Packages
import { getByTestId } from '@testing-library/react';
import { render } from 'jest-utils';

// @Project
import BaseLayout from '../index';

describe('BaseLayout component', () => {
  it('Should match snapshot', () => {
    const { container } = render(<BaseLayout />);

    expect(container).toMatchSnapshot();
  });

  it('Receives "children" prop', () => {
    const CUSTOMCHILDREN_TEST_ID = 'customchildren-testid';

    const { container } = render(
      <BaseLayout>
        <p data-testid={CUSTOMCHILDREN_TEST_ID}>Hola!</p>
      </BaseLayout>
    );

    const CustomChildrenNode = getByTestId(container, CUSTOMCHILDREN_TEST_ID);

    expect(CustomChildrenNode).toBeInTheDocument();
  })
});