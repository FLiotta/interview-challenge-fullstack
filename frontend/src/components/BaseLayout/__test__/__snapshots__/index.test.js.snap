// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout component Should match snapshot 1`] = `
<div>
  <div
    class="base-layout"
  >
    <div
      class="navbar"
    >
      <h6
        class="logo logo--brand-color"
        data-testid="logo-testid"
      >
        PiggyBank
      </h6>
      <div
        class="navbar__ctas"
      >
        <div
          class="profilebadge"
          data-testid="profilebadge-testid"
        >
          <p
            class="profilebadge__username"
          >
             
          </p>
          <div
            class="avatar"
            data-testid="avatar-testid"
            style="width: 30px; height: 30px;"
          >
            <div
              class="avatar__fill avatar__fill--1"
              data-testid="avatar-fill-testid"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="base-layout__body"
    />
  </div>
</div>
`;
